palace.pyx

```diff
@@ -832,6 +888,79 @@ cdef class Source:
         self.impl.destroy()
 
 
+cdef class SourceGroup:
+    cdef alure.SourceGroup impl
+
+    def __init__(self, context: Context = None) -> None:
```

Comment:
Forgot about this one, should have been `def __init__(self, context: Optional[Context]) -> None:`.  The class docstring should say what happens (or does not happen) if `context is None`.